capplets/about-me/mate-about-me.c

```diff
@@ -431,6 +477,15 @@ about_me_setup_dialog (void)
 	me->login = g_strdup (g_get_user_name ());
 	me->username = g_strdup (g_get_real_name ());
 
+#if HAVE_ACCOUNTSSERVICE
+	manager = act_user_manager_get_default ();
+	me->user = act_user_manager_get_user (manager, me->login);
+        g_object_get (manager, "is-loaded", &loaded, NULL);
+	if (!loaded) {
```

Comment:
Something does not look right here? You check `is-loaded` on manager, but then connect signal for user...

https://cgit.freedesktop.org/accountsservice/tree/src/libaccountsservice/act-user-manager.c#n2212
Says that trying to use this object before its `is-loaded` property is TRUE will result in undefined behavior.

So if `is-loaded` == `FALSE` you should not call `about_me_load_photo` or anything else that could use `me->user`.

I would do something like this:
```
...
on_user_is_loaded_changed (...)
{
  if (!act_user_is_loaded (user))
    return;

  about_me_load_photo (me);
  ...
}

...
me->user = act_user_manager_get_user (manager, me->login);
g_signal_connect (me->user, "notify::is-loaded", G_CALLBACK (on_user_is_loaded_changed), me);
on_user_is_loaded_changed (me->user, NULL, me);
...
```

Reply:
The change here should be right, the code comes from [GDM](https://github.com/GNOME/gdm/blob/master/daemon/gdm-manager.c#L1445).

If me->user is null, call about_me_load_photo will fallback to use ~/.face file.

Reply:
If CPU is slow to process, there will be no change when on_user_is_loaded_changed is called directly, `me->user` is still `NULL` at that time.

Reply:
Before use `act_user_manager_get_user()` to get user, need call `act_user_manager_list_users()` to loading the user list first. The previous modification called `act_user_manager_list_users()` directly, but without to use it, it looks like invalid code. 

So, now it's let the manager to load list automatically and finish it, then load image from accounts service, and then refresh the image on dialog.